code/modules/mob/living/bot/trashbot.dm

```diff
@@ -0,0 +1,142 @@
+/mob/living/bot/trashbot
+	name = "Trashbot"
+	desc = "A robot that picks up trash off of the floor."
+	icon_state = "trashbot0"
+	req_one_access = list(access_robotics, access_janitor)
+	botcard_access = list(access_janitor, access_maint_tunnels)
+
+	locked = 0 // Start unlocked so robot can set them to patrol, and turn them on.
+	wait_if_pulled = 1
+	min_target_dist = 0
+
+	var/idle_emote = "makes a quiet humming sound."
+
+	var/list/target_types = list()
+	var/list/stored_items = list()
+
+/mob/living/bot/trashbot/New()
+	..()
+	get_targets()
+	update_icons()
+
+/mob/living/bot/trashbot/handleIdle()
+	if(prob(5))
+		custom_emote(2, idle_emote)
+		playsound(src.loc, 'sound/machines/synth_yes.ogg', 50, 0)
+
+/mob/living/bot/trashbot/lookForTargets()
+	for(var/obj/D in view(world.view, src))
+		if(confirmTarget(D))
+			target = D
+			return
+
+/mob/living/bot/trashbot/confirmTarget(var/obj/D)
+	if(!..())
+		return 0
+	for(var/T in target_types)
+		if(istype(D, T))
+			return 1
+	return 0
+
+/mob/living/bot/trashbot/handleAdjacentTarget()
+	if(get_turf(target) == src.loc)
+		UnarmedAttack(target)
+
+/mob/living/bot/trashbot/UnarmedAttack(var/obj/D, var/proximity)
+	if(!..())
+		return
+
+	if(!istype(D))
+		return
+
+	if(D.loc != loc)
+		return
+
+	busy = 1
+	custom_emote(2, "sucks up \the [D].")
+	stored_items += D
+	del(D) //Not sure if a qdel would fuck anything up in this situation.
```

Comment:
If the bot is supposed to store the items it picks up, rather than just getting rid of them, you'll want to use forceMove() instead of del, and the del() proc should never be directly called, if I'm remembering things correctly.

Reply:
I was just going with, "Trashbot has a bluespace trashbag", because I'm not good with href or whatever that stuff is.

Reply:
You need to forceMove() the item into the bot. Right now it literally just destroys anything it picks up.

Reply:
Don't call del() on anything like ever ever ever. It's extremely slow.